ocaml/xapi/xapi_pool.ml

```diff
@@ -70,6 +70,14 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
       raise (Api_errors.Server_error(Api_errors.ha_is_enabled, []));
     end in
 
+  (* I cannot join a Pool if I have Clustering enabled on me *)
+  let clustering_is_not_enabled_on_me () =
+    let host = Helpers.get_localhost ~__context in
+    if not (Xapi_clustering.is_clustering_disabled_on_host ~__context host) then begin
+      error "Cannot join pool as Clustering is enabled";
+      raise (Api_errors.Server_error(Api_errors.clustering_enabled, []));
```

Comment:
Strictly speaking this error has a cluster_host parameter (in ocaml/idl/datamodel_errors.ml):
```ocaml
1127   error Api_errors.clustering_enabled ["cluster_host"]                                                                                                                      
1128     ~doc:"An operation was attempted while clustering was enabled on the cluster_host." ();
```
But it's not important as supplying that would require further changes.

Reply:
Minor suggestion: can just raise `Api_errors.(Server_error (clustering_enabled, [])) so you don't need the second `Api_errors`.

Also, I think it's worth providing the cluster_host parameter in the error message, though if you do that, you don't need to call `is_clustering_disabled_on_host` and can instead directly check the cluster_host's `enabled` field.

Reply:
I don't think there is much value in passing the cluster_host as the main way people will be disabling this is through XC which doesn't require this knowledge (Only way to do this is to untick "clustering_enabled" for the pool which then disables it for all hosts).

Reply:
If there errors are defined with arguments in the datamodel, then we should return those arguments, because clients (through the API bindings) may expect them.

Reply:
ok. Will make the update.